core/src/commonMain/kotlin/entity/User.kt

```diff
@@ -67,7 +96,8 @@ public open class User(
     /**
      * The complete user tag.
      */
-    public val tag: String get() = "$username#$discriminator"
+    @Suppress("DEPRECATION")
+    public val tag: String get() = if (migratedToNewUsernameSystem) "@$username" else "$username#$discriminator"
```

Comment:
i'm not sure about this one, should we do it like this or just deprecate it?

Reply:
[JDA](https://github.com/DV8FromTheWorld/JDA/pull/2462/files#diff-4c39d04c5713b45f7fed072d33c065313d40af373e8b6513516ede58a94da8b8) seems to deprecate it, [discord.js](https://github.com/discordjs/discord.js/pull/9512/files#diff-b7b221539f62dcbc8c29a6cad99d732320243e3bcc5fab3e4fec4383ac4d6bd1) seems to do it like i did here

Reply:
I'd go with a straight deprecation and add another Val that gets the @username because the at username has a different name now if I remember correctly. I just don't remember said name

Reply:
> Val that gets the @username because the at username has a different name now if I remember correctly. I just don't remember said name

It's still called a [username](https://support.discord.com/hc/en-us/articles/12620128861463#h_01GXPQAGG6W477HSC5SR053QG1) :smile:

Reply:
Hello
I'd go with deprecation
It's possible that Discord will give us a window to migrate given that the username change is rolling out incrementally.

Reply:
update: discord.js also seems to deprecate it: https://github.com/discordjs/discord.js/pull/9512/commits/22b743f861855d64a0672d038de6c9fc98fea965